server/src/main/java/org/jboss/as/server/operations/AbstractInstallationReporter.java

```diff
@@ -0,0 +1,303 @@
+
+    /**
+     * Create a ModelNode representing the operating system the instance is running on.
+     *
+     * @return a ModelNode representing the operating system the instance is running on.
+     * @throws OperationFailedException
+     */
+    private ModelNode createOSNode() throws OperationFailedException {
+        String osName = getProperty("os.name");
+        final ModelNode os = new ModelNode();
+        if (osName != null && osName.toLowerCase().contains("linux")) {
+            try {
+                os.set(GnuLinuxDistribution.discover());
+            } catch (IOException ex) {
+                throw new OperationFailedException(ex);
+            }
+        } else {
+            os.set(osName);
+        }
+        return os;
+    }
+
+    /**
+     * Create a ModelNode representing the JVM the instance is running on.
+     *
+     * @return a ModelNode representing the JVM the instance is running on.
+     * @throws OperationFailedException
+     */
+    private ModelNode createJVMNode() throws OperationFailedException {
+        ModelNode jvm = new ModelNode().setEmptyObject();
+        jvm.get(NAME).set(getProperty("java.vm.name"));
+        jvm.get(JAVA_VERSION).set(getProperty("java.vm.specification.version"));
+        jvm.get(JVM_VERSION).set(getProperty("java.version"));
+        jvm.get(JVM_VENDOR).set(getProperty("java.vm.vendor"));
+        jvm.get(JVM_HOME).set(getProperty("java.home"));
+        return jvm;
+    }
+
+    /**
+     * Create a ModelNode representing the CPU the instance is running on.
+     *
+     * @return a ModelNode representing the CPU the instance is running on.
+     * @throws OperationFailedException
+     */
+    private ModelNode createCPUNode() throws OperationFailedException {
+        ModelNode cpu = new ModelNode().setEmptyObject();
+        cpu.get(ARCH).set(getProperty("os.arch"));
+        cpu.get(AVAILABLE_PROCESSORS).set(Runtime.getRuntime().availableProcessors());
+        return cpu;
+    }
+
+    /**
+     * Create a ModelNode representing the instance product and the plateform it is running on.
+     *
+     * @param context the operation context.
+     * @param environment the current runtime environment.
+     * @param patchingInfo
+     * @param installationDir installation directory of the current server instance.
+     * @return a ModelNode representing the instance product and the plateform it is running on.
+     * @throws OperationFailedException
+     */
+    protected ModelNode createProductNode(OperationContext context, ProductEnvironment environment,
+            ModelNode patchingInfo, Path installationDir) throws OperationFailedException {
+        assert environment != null;
+        ModelNode product = new ModelNode().setEmptyObject();
+        product.get(REPORT_VERSION).set("1.0");
+        product.get(NODE_NAME).set(environment.getHostName());
+        product.get(HOSTNAME).set(environment.getQualifiedHostName());
```

Comment:
environment.getQualifiedHostName() may return the values of system properties jboss.qualified.host.name or jboss.host.name, and not necessarily some OS type concept such as the HOSTNAME or COMPUTERNAME environment variables or the result of something like InetAddressUtil.getLocalHost().getHostName().

I'm not sure if that WildFly-specific deviation from any "cross-product" meaning of "Host Name" matters or not.

Using -Djboss.host.name is a common way to assign the name for a host in the management model without having to hard code it in each host's host.xml, so this isn't an edge case.